src/artifacts_poetry_plugin/plugin.py

```diff
@@ -9,143 +9,234 @@
+        project_package_path = os.path.join(poetry.root_dir, "dist")
+        project_package = poetry.package
+        for root, dirs, files in os.walk(project_package_path):
+            for file in files:
+                if file.endswith(".whl") or file.endswith(".tar.gz"):
+                    file_data = {"file": file, "url": Path(os.path.join(root, file))}
+                    project_package.files.append(file_data)
+        return project_package
+
+    def get_all_poetry(self, path, downstream):
+        poetry_files = []
+        if downstream is False:
+            poetry = self.create_poetry(path, os.listdir(path))
+            if poetry is not None:
+                poetry_files.append(poetry)
+                return poetry_files
+        for root, dirs, files in os.walk(Path(path)):
+            poetry = self.create_poetry(root, files)
+            if poetry is not None:
+                poetry_files.append(poetry)
+        return poetry_files
+
+    def create_poetry(self, root, files):
+        poetry = None
+        if "pyproject.toml" in files and "poetry.lock" in files:
+            poetry_root = Path(root)
+            poetry = Factory().create_poetry(poetry_root)
+            poetry.root_dir = poetry_root
+            logger.info(f"Found python project at {root}")
+        return poetry
+
+    def log_non_deployable_packages(self, poetry_files):
+        package_set = set()
+        for poetry in poetry_files:
+            for package in poetry.non_deployable_packages:
+                if package.name not in package_set:
+                    package_set.add(package.name)
+                    logger.warn(
+                        f"Could not find any wheel or tar.gz files for package {package.name}:{package.pretty_version}. Package may be statically linked or doesnt exist in artifacts cache"
+                    )
+
+    def create_poetry_packages(self, root_dir, cache_dir, deploy_downstream):
+        """Gets all poetry packages required by each downstream project from the given cache directory."""
+        poetry_files = []
+        poetry_files = self.get_all_poetry(root_dir, deploy_downstream)
+        for poetry in poetry_files:
+            packages = self.get_poetry_packages(poetry)
+            wheel_files = self.get_cached_wheel_files(cache_dir)
+            (
+                deployable_packages,
+                non_deployable_packages,
+            ) = self.get_deploy_non_deploy_packages(
+                packages=packages, wheel_files=wheel_files
+            )
+            project_package = self.get_project_package(poetry)
+            if len(project_package.files) > 0:
+                deployable_packages.append(project_package)
+            else:
+                non_deployable_packages.append(project_package)
+            poetry.deployable_packages = deployable_packages
+            poetry.non_deployable_packages = non_deployable_packages
```

Comment:
what exactly is the difference between deployable and non deployable packages?

Reply:
deployable means that the plugin is going to deploy the package to the alternate repository. Non deployable means it is not able to. The non deployable packages are logged.